api/encoding/call_test.go

```diff
@@ -46,6 +48,20 @@ func TestNilCall(t *testing.T) {
 	assert.Equal(t, "", call.OriginalHeader("foo"))
 	assert.Empty(t, call.HeaderNames())
 	assert.Nil(t, call.OriginalHeaders())
+	assert.Equal(t, 0, call.HeadersLen())
+	assert.Equal(t, 0, call.OriginalHeadersLen())
+	assert.Len(t, slices.Collect(call.HeaderNamesAll()), 0, "nil call should yield no headers")
+	headersAll := make(map[string]string)
+	for k, v := range call.HeadersAll() {
+		headersAll[k] = v
+	}
```

Comment:
```suggestion
	headersAll := maps.Collect(calls.HeadersAll())
```

(and below 4 times too)

we're doing this for header names, why not be consistent?